erts/emulator/sys/unix/sys_drivers.c

```diff
@@ -1814,29 +1824,37 @@ static ErlDrvSSizeT forker_control(ErlDrvData e, unsigned int cmd, char *buf,
         first_call = 0;
     }
 
-    driver_enq(port_num, buf, len);
-    if (driver_sizeq(port_num) > sizeof(*proto)) {
-        return 0;
-    }
-
-    if ((res = sys_uds_write(forker_fd, (char*)proto, sizeof(*proto),
-                             proto->u.start.fds, 3, 0)) < 0) {
-        if (errno == ERRNO_BLOCK || errno == EINTR) {
-            driver_select(port_num, forker_fd, ERL_DRV_WRITE|ERL_DRV_USE, 1);
-            return 0;
-        } else if (errno == EMFILE) {
-            forker_sigchld(proto->u.start.port_id, errno);
-            forker_deq(port_num, proto);
+    if (proto->action == ErtsSysForkerProtoAction_Start) {
+        driver_enq(port_num, buf, len);
+        if (driver_sizeq(port_num) > sizeof(*proto)) {
             return 0;
-        } else {
-            erts_exit(ERTS_DUMP_EXIT, "Failed to write to erl_child_setup: %d\n", errno);
         }
-    }
+
+        if ((res = sys_uds_write(forker_fd, (char*)proto, sizeof(*proto),
+                                proto->u.start.fds, 3, 0)) < 0) {
+            if (errno == ERRNO_BLOCK || errno == EINTR) {
+                driver_select(port_num, forker_fd, ERL_DRV_WRITE|ERL_DRV_USE, 1);
+                return 0;
+            } else if (errno == EMFILE) {
+                forker_sigchld(proto->u.start.port_id, errno);
+                forker_deq(port_num, proto);
+                return 0;
+            } else {
+                erts_exit(ERTS_DUMP_EXIT, "Failed to write to erl_child_setup: %d\n", errno);
+            }
+        }
 
 #ifndef FORKER_PROTO_START_ACK
-    ASSERT(res == sizeof(*proto));
-    forker_deq(port_num, proto);
+        ASSERT(res == sizeof(*proto));
+        forker_deq(port_num, proto);
 #endif
+    } else if (proto->action == ErtsSysForkerProtoAction_Stop) {
+        if ((res = write(forker_fd, (char*)proto, sizeof(*proto))) < 0) {
+            if (errno != EAGAIN && errno != EINTR) {
```

Comment:
These are a bit of a random guess.  I saw EAGAIN come up in some test cases and found other guard conditions allowing these errors to be non-fatal.  That vaguely makes sense, because they have to do with the `write` syscall failing on an otherwise operational pipe.

It might be worthwhile to retry the write later?  But the precedent I see elsewhere in the code seems to be more practical, this message is a nice-to-have at the moment.